daft/runners/partitioning.py

```diff
@@ -92,7 +92,8 @@ def __post_init__(self) -> None:
                 raise ValueError(f"mismatch of column id: {col_id} vs {tile.column_id}")
 
     def __len__(self) -> int:
-        assert len(self.columns) > 0
```

Comment:
@samster25 what was this assert meant for? is it okay to return 0 in this case?

Reply:
Before I was thinking that an vPartition of len 0 would be made up of columns that were len 0, but this makes sense as well. I can think of a potential bug though for compute, what if you are reading from a directory of CSVs and one of the files is header-only. Would this still work?

Reply:
Not sure I fully understand the potential failure case here :sweat_smile:  if the file is header-only and read into a partition, would it still be correct to return the length as zero?

Reply:
I ended up writing a test for my usecase here: https://github.com/Eventual-Inc/Daft/pull/343
Looks like my edge case doesn't really matter since you can't run compute on those columns since they don't exist.  Everything else looks good! :) Great first PR.